Source/TestDafny/MultiBackendTest.cs

```diff
@@ -89,6 +94,11 @@ public class TestDafny {
     var success = true;
     foreach (var plugin in dafnyOptions.Plugins) {
       foreach (var compiler in plugin.GetCompilers(dafnyOptions)) {
+        if (compiler.TargetId == "lib") {
+          // Some tests still fail when using the lib back-end, for example due to disallowed assumptions being present in the test,
+          // Such as empty constructors with ensures clauses, generated from iterators
+          continue;
+        }
```

Comment:
Ah this is how you kept it all green :) This is reasonable for now, but it'll be good to dig into those tests to see if the other backends are too permissive about this.